tools/update.php

```diff
@@ -0,0 +1,705 @@
+    // and the latest version, add the relevant patches if they begin with
+    // a number in that range. For example, if upgrading from 17 to 19, the
+    // array will contain all aptches beginning with "17", "18", or "19".
+    // i.e.:
+    //   [0] => /var/www/loris/SQL/Release_patches/17.0_To_18.0_upgrade_A.sql
+    //   [1] => /var/www/loris/SQL/Release_patches/17.0_To_18.0_upgrade_B.sql
+    //   [2] => /var/www/loris/SQL/Release_patches/18.0_To_19.0_upgrade.sql
+    //   [3] => /var/www/loris/SQL/Release_patches/19.0_To_19.1_upgrade.sql
+    // Range will run once when $start = $end. We want this in the case where
+    // there is no difference in major versions but there is a change in minor
+    // versions. The below loops will, in this case, match the minor release
+    // patches.
+    $start = $from_versions[MAJOR];
+    $end   = $start; // default case: there is a minor release difference
+    if ($diff_major > 0) {
+        $end = $to_versions[MAJOR];
+    }
+    $patch_dir           = $loris_root . 'SQL/Release_patches/';
+    $all_release_patches = glob($patch_dir . '*.sql');
+    $patches = [];
+    if ($diff_major > 0 || $diff_minor > 0) {
+        foreach (range($start, $end) as $v) {
+            foreach ($all_release_patches as $patch) {
+                // strpos must be 0 or else this will match w/ e.g. 16.0_To_17.0
+                // when the server is already on version 17
+                if (strpos(basename($patch), "$v") === 0) {
+                    $patches[] = $patch;
+                }
+            }
+        }
+    }
+    if (isDev()) {
+        echo "[+] Developer instance detected. Including developer patches..."
+            . PHP_EOL;
+
+        // Add all patches in Archive/$MAJOR.$MINOR. Everything other needed
+        // command will be in the Release patches which are been added above
+        $dev_patch_dir = $loris_root . 'SQL/Archive/' . $to_versions[MAJOR]
+            . '.' . $to_versions[MINOR] . '/';
+        $patches       = array_merge($patches, glob($dev_patch_dir . '*.sql'));
+    }
+    return $patches;
+}
+
+/**
+ * Apply a list of SQL patch files using exec.  Currently this function is only
+ * called with $report_only = true so patches are never applied. This can be
+ * changed in the future if this functionality is required.
+ *
+ * @param array      $patches     Patch files to be applied
+ * @param dictionary $db_config   DB config info. Needed for credentials
+ * @param bool       $report_only Whether to only display patches or to also
+ *                          apply them. Always true (for now).
+ *
+ * @return bool Always true (for now). False if patches fail to be applied.
+ */
+function applyPatches($patches, $db_config, $report_only = true) : bool
+{
+    // Iterate over all patches and source them into MySQL
+    $A = $db_config['database'];
+    $u = $db_config['username'];
+    $p = $db_config['password'];
+    $h = $db_config['host'];
```

Comment:
descriptive variable names would be better

Reply:
They are descriptive to the extent that they match the flags for `mysql` on the command line.